src/norm_oracles.jl

```diff
@@ -360,3 +360,64 @@ function compute_extreme_point(lmo::EllipsoidLMO, direction; v=nothing, kwargs..
     mul!(v, I, lmo.buffer, -scaling, true)
     return v
 end
+
+
+"""
+    OrderWeightNormLMO(weights,radius)
+    
+LMO with feasible set being the atomic ordered weighted l1 norm: https://arxiv.org/pdf/1409.4271
+
+```
+C = {x ∈ R^n, Ω_w(x) ≤ R} 
+```
+The weights are assumed to be positive.
+"""
+struct OrderWeightNormLMO{R,B,D} <: LinearMinimizationOracle
+    radius::R
+    mat_B::B
+    direction_abs::D
+end
+
+function OrderWeightNormLMO(weights, radius)
+    N = length(weights)
+    B = zeros(N)
+    s = zero(eltype(weights))
+    w_sort = sort(weights,rev=true)
+    for i in 1:N
+        s += w_sort[i]
+        B[i] = 1/s
+    end
+    direction_abs = similar(weights)
+    return OrderWeightNormLMO(radius,B,direction_abs)
+end
+
+function compute_extreme_point(
+    lmo::OrderWeightNormLMO,
+    direction::M;
+    v=nothing,
+    kwargs...,
+) where {M}
+    for i in eachindex(direction)
+        lmo.direction_abs[i] = abs(direction[i])
+    end
+    perm_grad = sortperm(lmo.direction_abs,rev=true)
+    max = 0
+    ind = 1
+    N = length(lmo.mat_B)
+    for i in 1:N
+        scal = 0
+        for k in 1:i
+            scal += lmo.mat_B[i]*(lmo.direction_abs[perm_grad])[k]
+        end
```

Comment:
this will be type-unstable (since scal is an Int only if no iteration is performed)